controllers/chat/group/groupAddMembers.js

```diff
@@ -57,6 +61,25 @@ const groupAddMembers = async (req, res) => {
   try {
     const responseChannel = await channelToAdd?.addMembers(filteredMembersObject);
 
+    channelMembers.map(async (member) => {
+      const targetUserModel = await UsersFunction.findUserById(User, member);
+
+      const textOwnUser = `You added ${targetUserModel.username} to this group`;
+      const textTargetUser = `${ownUser.username} added you to this group`;
+      const textDefaultUser = `${ownUser.username} added ${targetUserModel.username} to this group`;
+
+      await channelToAdd.sendMessage({
+        text: textDefaultUser,
+        own_text: textOwnUser,
+        other_text: textTargetUser,
+        type: 'system',
+        user_id: userId,
+        system_user: userId,
+        isSystem: true,
+        members: [member]
+      });
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code sends a system message for each member added to the group. However, it's inside a `map` function which doesn't handle asynchronous operations well. It would be better to use `Promise.all` with `map` to ensure all promises are resolved before proceeding. Also, error handling should be added to catch any errors that might occur during the sending of messages.
```diff
- channelMembers.map(async (member) => {
+ Promise.all(channelMembers.map(async (member) => {
    const targetUserModel = await UsersFunction.findUserById(User, member);
  
    const textOwnUser = `You added ${targetUserModel.username} to this group`;
    const textTargetUser = `${ownUser.username} added you to this group`;
    const textDefaultUser = `${ownUser.username} added ${targetUserModel.username} to this group`;
  
-   await channelToAdd.sendMessage({
+   return channelToAdd.sendMessage({
      text: textDefaultUser,
      own_text: textOwnUser,
      other_text: textTargetUser,
      type: 'system',
      user_id: userId,
      system_user: userId,
      isSystem: true,
      members: [member]
    });
- });
+ })).catch(error => {
+   console.error('Error sending system messages:', error);
+ });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->